### PR TITLE
Modify masterbar module to include new unified admin color schemes file

### PR DESCRIPTION
**Pull request by getdave**
Alternative to https://github.com/Automattic/jetpack/pull/17750/.

As discussed in p1604929101083300-slack-ajax this PR repurposes and extends the existing Masterbar module to become a more all purpose module for any customizations required to the WPAdmin Dashboard experience on WP.com.

Specifically, where there was previously only the `A8C_WPCOM_Masterbar` class, we now add an additional `A8C_WPCOM_Unified_Admin_Color_Schemes` class and init that from the main file as well.

In the future, other classes could be added.

Please note that @obenland has already started work on customising the masterbar module here so we could look to land that. However, as it stands my PR (ie: this one) contains less code and is more "bare bones" and thus might be easier to merge.

#### Changes proposed in this Pull Request:

* Add `A8C_WPCOM_Unified_Admin_Color_Schemes` class to masterbar module and init from main file.
* Repurpose master bar module to become an all-purpose module for any customizations required to the WPAdmin Dashboard experience on WP.com.
* Add PHP `namespace` to the module.
* Move individual concerns into separate directories (eg: `/masterbar`, `/unified-admin-color-schemes`...etc).

#### Jetpack product discussion

* paYJgx-17t-p2#comment-1391
* p1604929101083300-slack-ajax

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout PR
* Modify `init` method of `modules/masterbar/unified-admin-color-schemes.php` to include some debug output.
* Check that debug output is shown _only_ on admin pages.
* Verify existing masterbar works as previously.

#### Proposed changelog entry for your changes:
* Repurpose master bar module to become an all-purpose module for any customizations required to the WPAdmin Dashboard experience on WP.com.
* Add unified color schemes `class` to masterbar module and init from main file.
* Add PHP `namespace` to the module.
* Move individual concerns into separate directories (eg: `/masterbar`, `/unified-admin-color-schemes`...etc).

### CODE REVIEW

File: modules/masterbar/unified-admin-color-schemes.php
Comment (jeherve) [inline modules/masterbar/unified-admin-color-schemes.php:1]: Maybe we can take this opportunity to fix that, use a namespace for all Masterbar work, a simpler class name, and a file name that matches it?

Comment (getdave) [reply modules/masterbar/unified-admin-color-schemes.php]: Added a namespace 

```
namespace Automattic\Jetpack\Dashboard_Customizations;
```

Is this what you had in mind? Is that ok PHP compact wise with Jetpack?

Comment (jeherve) [reply modules/masterbar/unified-admin-color-schemes.php]: Yes, exactly, thank you!

File: modules/masterbar/masterbar/masterbar.php
Comment (getdave) [inline modules/masterbar/masterbar/masterbar.php:173]: I'm not sure on the deal here. Seems PHPCS automatically "fixed" this for me but is it right? Surely we don't want to change the action?

Comment (getdave) [reply modules/masterbar/masterbar/masterbar.php:173]: Note this happens in _several_ places in this file.

Comment (marekhrabe) [reply modules/masterbar/masterbar/masterbar.php:173]: If you put the prefix back, does it get removed again?

Comment (getdave) [reply modules/masterbar/masterbar/masterbar.php]: Fixed.

**Comment by marekhrabe** [comment]
This works well for me and the testing `echo` I've done in the color scheme class has worked in wp-admin correctly ✅ 

I _think_ we are missing one last step and that is updating the module definition in `modules/module-headings.php`, since the name and description were updated. `yarn build` should get you sorted and you should commit the updated file.

**Review by marekhrabe — DISMISSED** [review]

**Comment by getdave** [comment]
Just waiting on Crew to approve.

**Review by jeherve — DISMISSED** [review]
👍 🚢 

> Just waiting on Crew to approve.

Just noting that @marekhrabe can now approve PRs too. @marekhrabe, if you're happy with a PR and don't think an extra crew review is necessary, you can flip the labels to "ready to merge" and y'all can merge things right away.

File: modules/masterbar.php
Comment (obenland) [inline modules/masterbar.php:16]: Does it matter if this is `require` or `require_once`? Jetpack seems to support `__DIR__ `, in case that is something you want to change.

File: modules/masterbar.php
Comment (obenland) [inline modules/masterbar.php:20]: Could this just be `Admin_Color_Schemes`?

File: modules/masterbar.php
Comment (getdave) [reply modules/masterbar.php:16]: I wanted to stick as closely to what was in the existing code as possible. Happy to amend in the followup.

File: modules/masterbar.php
Comment (getdave) [reply modules/masterbar.php:20]: Given the x2 👍 how strongly do you feel about that?

Comment (obenland) [reply modules/masterbar.php:20]: Pretty strongly, but we can also just do it when it actually gets implemented.

File: modules/masterbar/masterbar/masterbar.php
Comment (obenland) [inline modules/masterbar/masterbar/masterbar.php:228]: @jeherve Is `RTL_Admin_Bar` something that we need at all? It looks like it's currently defunct, the `wp_admin_bar_class` callback gets registered on `admin_bar_init`, which is after `wp_admin_bar_class` runs.
Testing with Hebrew, the master bar seems to render fine without it.

File: modules/masterbar.php
Comment (getdave) [reply modules/masterbar.php:20]: Done

File: modules/masterbar/masterbar/masterbar.php
Comment (getdave) [reply modules/masterbar/masterbar/masterbar.php:231]: Appreciate this may be something we want to look at. However, the aim of this PR is really only to get a 1:1 parity between what we already have for masterbar and the version of masterbar in this PR. The only difference should be that we are now making the module more generic and will be in a better place to add more features such as admin color schemes.

In order to unblock the EU timezone folks, I'm going to raise an Issue to handle this in a followup.

Comment (getdave) [reply modules/masterbar/masterbar/masterbar.php:231]: Followup Issue https://github.com/Automattic/jetpack/issues/17773

File: modules/masterbar.php
Comment (getdave) [reply modules/masterbar.php:16]: Followup https://github.com/Automattic/jetpack/issues/17774

File: modules/masterbar/masterbar/masterbar.php
Comment (getdave) [reply modules/masterbar/masterbar/masterbar.php:231]: Follow up https://github.com/Automattic/jetpack/issues/17773

**Review by marekhrabe — APPROVED** [review]
I've rested everything again with a code confirmed to be run in the color schemes class. With remaining issues externalized, I think we should be good proceeding and have a base for other color scheme tasks.

**Comment by getdave** [comment]
FYI anyone who comes to this and wonders what happened to syncing this code via Fusion please see

p1605115584001400-slack-ajax-reports

> we can sync the code later if we need to (it's currently not synced by fusion)

**Comment by getdave** [comment]
Thanks for merging @obenland. I really appreciate that.

**Pull request merged**
